syntax = "proto2";

message AuthenticateArgs {
  optional string gameId = 1;
  optional string userId = 2;
}

message AuthenticateOutput {
  optional string token = 1;
  optional string userId = 2;
  optional bool showBranding = 3 [default = false];
  optional string playerInsightState = 4;
  optional bool isSocialNetworkUser = 5 [default = false];
  optional bool isInstalledByPublishingNetwork = 6 [default = false];
  optional bool deprecated1 = 7 [default = false];
  optional string apiSecurity = 8;
  repeated string apiServerHosts = 9;
}

message KeyValuePair {
  optional string key = 1;
  optional string value = 2;
}

message CreateJoinRoomArgs {
  optional string roomId = 1;
  optional string roomType = 2;
  optional bool visible = 3;
  repeated KeyValuePair roomData = 4;
  repeated KeyValuePair joinData = 5;
  optional bool isDevRoom = 6;
}

message CreateJoinRoomOutput {
  optional string roomId = 1;
  optional string joinKey = 2;
  repeated ServerEndpoint endpoints = 3;
}

message CreateJoinRoomError {
  optional string message = 1;
  optional int32 errorCode = 2;
}

message ServerEndpoint {
  optional string address = 1;
  optional int32 port = 2;
}